extensions/guacamole-auth-cas/src/main/resources/casConfig.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Config block which registers CAS-specific field types.
+ */
+angular.module('guacCAS').config(['formServiceProvider',
+        function guacCASConfig(formServiceProvider) {
+
+    // Define field for ticket from CAS service
+    formServiceProvider.registerFieldType("GUAC_CAS_TICKET", {
+        templateUrl   : '',
+        controller    : 'guacCASController',
+        module        : 'guacCAS'
+    });
+
+}]);
+
+/**
+ * Config block which augments the existing routing, providing special handling
+ * for the "ticket=" parameter provided by the CAS authentication process.
+ */
+angular.module('index').config(['$routeProvider','$windowProvider',
+        function indexRouteConfig($routeProvider,$windowProvider) {
```

Comment:
1) Okay, I'll figure out some other block type to move it to.
2) Yeah.

No, the block of code you highlighted there falls into the hole I mentioned before.  Specifically:
- The comment indicates that it transforms /#/id_token= to /#/?id_token=.  The situation I end up with is that I need to transform ?id_token=/#/ to /#/?id_token=.
- Because the ?id_token= parameter gets placed before the /#/, the $routeProvider.when block never actually executes, because the $routeProvider code only ever looks at anything passed the hash tag.
- I can't remember for sure if this was true or not, but I believe that this also means that the $location.search(params) line also does not work, because it's looking at anything trailing the #/ on the URL, not what occurs in front of it.

I'm happy to continue trying stuff to find a better solution to this - let me know if you have other suggestions!